### ISSUE TITLE
[New] Lightning Terminal

### ISSUE THREAD

**Issue by openoms** [raspiblitz/raspiblitz#1993]
https://github.com/lightninglabs/lightning-terminal will replace the separate Loop, Pool and Faraday installations.

Find out more about usage: https://github.com/lightninglabs/lightning-terminal/blob/master/doc/WALKTHROUGH.md

To test:
```
# remove the previous version of the script:
rm -f bonus.lit.sh
# download:
wget https://raw.githubusercontent.com/openoms/raspiblitz/new-lit/home.admin/config.scripts/bonus.lit.sh
# check:
cat bonus.lit.sh
# install:
bash bonus.lit.sh on
# start service:
sudo systemctl start litd
# info:
bash bonus.lit.sh menu
```

~~If you have used Pool, Loop and Faraday before now there will be duplicate instances under separate users. The Lightning Terminal won't see your old Loop and Pool transactions and accounts.~~
UPDATE with https://github.com/rootzoll/raspiblitz/pull/1994/commits/910dcf5466b119514c99de3e0f58fc7da9f2220b the data directories of Loop and Pool are symlinked to LiT and previous data will appear.
Reinstall with the updated script to take effect if missing Pool data especially.

**Comment by Omicron59** [comment]
Hello openoms  I tested the new terminal last night, really great working. I rebooted my raspiblitz and since it is blocked on lightning progress and I have an error, will you have an idea, or it is normal that it is so long. thanks to you

![image](https://user-images.githubusercontent.com/46853496/106382065-460a9700-63bd-11eb-9d09-58ad2e671511.png)

**Comment by openoms** [comment]
@mamar59 your issue i not Lightning Terminal related. Seems the same as: https://github.com/rootzoll/raspiblitz/issues/1982, please discuss there. 
Looking at the output of the logs with `debug` can help a lot with details.

**Comment by nyxnor** [comment]
UI password is presented on the logs. just a warn to anyone who post the output to this `sudo systemctl status litd`.
Openoms, I needed to restart (useful to be in the instructions), because stopping tor stopped all other services, so starting one by one was not optimal.
@mamar59 Have you tried the UI? Try the lightning pool screen to see if you get the same thing.

Ligthing pool segment, it´s saying the token was incorrect and I needed to `listauth` to check.
![image](https://user-images.githubusercontent.com/69700936/106388367-ef9d5880-63bc-11eb-9a80-d1812e51639a.png)

using `sudo systemctl status litd`
![image](https://user-images.githubusercontent.com/69700936/106387442-83b8f100-63b8-11eb-8681-df4d9030c4e6.png)
![image](https://user-images.githubusercontent.com/69700936/106387447-93383a00-63b8-11eb-8387-b5571d70f0fb.png)

EDIT:
Found this https://github.com/lightninglabs/lightning-terminal/issues/117#issuecomment-691400785
And this https://github.com/lightninglabs/lightning-terminal/issues/117#issuecomment-692304583
And this https://github.com/lightninglabs/lightning-terminal/issues/139#issuecomment-723581402

**Comment by openoms** [comment]
@Ben10Brasileiro indeed running two instances of Pool can cause this issue, so in the latest version of the script Pool and Loop is uninstalled and the data is moved to LiT. 

The uipassword can be moved to a config file, that is to be done.

You will need to go to the command line to find the token and remove it manually. It is complicated with LiT now before we add the aliases. You can remove LiT and install Pool temporarily to use the `listauth` command and remove the token.

Then when you reinstall with the latest script it should work, but indeed all to be tested still.

**Comment by openoms** [comment]
OK, switched to the config file from storing parameter in the systemd service in https://github.com/rootzoll/raspiblitz/pull/1994/commits/e45f34652659fd6dc2c2567439406ad8500db902.

Aliases are added with https://github.com/rootzoll/raspiblitz/pull/1994/commits/c7579bce2199250269c8a24bd9fde3ddd4fdeff6

A reinstall with the latest script is recommended or add the aliases:
```
source /mnt/hdd/raspiblitz.conf
  echo "
alias lit-loop=\"loop --rpcserver=localhost:8443 \\
  --tlscertpath=~/.lit/tls.cert \\	
  --macaroonpath=~/.loop/${chain}net/loop.macaroon\"
alias lit-pool=\"pool --rpcserver=localhost:8443 \\
  --tlscertpath=~/.lit/tls.cert \\	
  --macaroonpath=~/.pool/${chain}net/pool.macaroon\"
alias lit-frcli=\"frcli --rpcserver=localhost:8443 \\
  --tlscertpath=~/.lit/tls.cert \\
  --macaroonpath=~/.faraday/${chain}net/faraday.macaroon\"
" | sudo tee -a /home/lit/.bashrc
```

Sample info menu now:
![image](https://user-images.githubusercontent.com/43343391/106393222-d2cd4900-63ed-11eb-80cb-f20f3d275c6e.png)


Still to be done is:
- [ ] SERVICES menu integration (will remove the standalone Pool, Loop, Faraday)
- [ ] integrate in recovery
- [ ] update when password_B is changed
- [ ] Test Loop in RTL with LiT (likely needs conf changes)

**Comment by openoms** [comment]
When getting this error:
![image](https://user-images.githubusercontent.com/43343391/107086013-42c63f80-67f1-11eb-9e9f-5beacadd2abf.png)

show the file with:
```
sudo su - lit
lit-pool listauth
```
to remove:
```
rm /home/lit/.pool/mainnet/lsat.token
```

**Comment by nyxnor** [comment]
I tried with loop and pool installed and uninstalled...
Did the steps on the first post to remove and reinstall lit too.
I have to say it worked for the first time, it was not appearing the same error, was another error that I cant show again cause its saying my password is wrond, even though lnd, litd, nginx is running and I have already restarted and reboote multiple times.

![image](https://user-images.githubusercontent.com/69700936/107096814-4aceb100-67ea-11eb-8958-da383f50cf4d.png)

**Comment by openoms** [comment]
@Ben10Brasileiro to gevrid of that erroneous entry uninstall lit first with:
```
bash bonus.lit.sh off
```
then reinstall:
```
bash bonus.lit.sh on
```
This is the same as reinstalling from the SERVICES menu when it is implemented.

**Comment by nyxnor** [comment]
@openoms Did not worf :(
Ive flashed the sd card and deleted all but the blockchain recently, tried the first steps to install, but was appearing the warn on pool screen even tough I didnt install pool this time. So removing the file was not appearing as there was non. Did off and on, but still same result.
I installed pool, removed the token from here `/home/pool/.pool/mainnet/lsat.token.pending`, checked that removed with ls and ON for litd and START, but still same result.
Then I found it on `/mnt/hdd/app-data/.pool/mainnet/lsat.token.pending`, deleted from here too, checked and it had deleted, but appeared seconds after.

![image](https://user-images.githubusercontent.com/69700936/107866626-15be0080-6e73-11eb-85d9-748f52368337.png)

Can I remove the following directory? `/mnt/hdd/app-data/.pool`

**Comment by openoms** [comment]
@Ben10Brasileiro did you follow these steps?
https://github.com/rootzoll/raspiblitz/issues/1993#issuecomment-774273660 ?

what does `lit-pool listauth` show?

**Comment by nyxnor** [comment]
@openoms Yes, I had tried that before
![image](https://user-images.githubusercontent.com/69700936/107877219-5184b480-6ecb-11eb-8fac-e9b421b78ef2.png)

**Comment by openoms** [comment]
>Ive flashed the sd card and deleted all but the blockchain recently, tried the first steps to install, but was appearing the warn on pool screen even tough I didnt install pool this time. So removing the file was not appearing as there was non. Did off and on, but still same result.
I installed pool, removed the token from here /home/pool/.pool/mainnet/lsat.token.pending, checked that removed with ls and ON for litd and START, but still same result.

@Ben10Brasileiro you might need to start with a fresh SDcard again. 
To be clear LiT does include Pool so if that warning comes up you need to use `lit-pool listauth` to find the and remove the token.
DO not install install Pool on top of LiT beacuse then you have a duplicate of the service causing problems.

**Comment by nyxnor** [comment]
I did that at firs, but Im going to detail what I did for this results https://github.com/rootzoll/raspiblitz/issues/1993#issuecomment-778712622

Saving this history to remember what does not work.
1 Cleaned all but blockchain and flashed sd card.
2 Installed `lit` and tried sudo su - lit & lit-pool listauth & rm /home/lit/.pool/mainnet/lsat.token, but it was saved as lsat.token.pending, wich I deleted but appeared seconds after.
3 Remove `lit` and Installed `pool` as instructed here https://github.com/rootzoll/raspiblitz/issues/1993#issuecomment-770402950 to to listauth as pool user, found the token here `/home/pool/.pool/mainnet/lsat.token.pending`, after removing with `sudo rm -rf xxx`, it was removed, checked with `ls`, but appeared some seconds after.
4 Uninstalled `pool`, and tried again to remove the token with lit user, same result, it deleted but appeared seconds after.
5 Removing /mnt/hdd/app-data/.pool didnt work, gave more erros on the lit terminal.

**Comment by nyxnor** [comment]
I dont have nothing on Pool, altough they are talking about Loop above.
I think that with 0 sat, the listauth warning will keep appearing, need to fund first to test if it's the problem.


![image](https://user-images.githubusercontent.com/69700936/108006814-ba1f7e80-6ffc-11eb-973e-5e88d963d226.png)

**Comment by openoms** [comment]
Good find! Do you have anything in LN channels? I did not try on a fresh node yet.

**Comment by nyxnor** [comment]
No, nothing yet, need to understand coinjoin and coinjoin first :)

**Comment by openoms** [comment]
So the takeaway is that you need to be able to pay 1 sat (so have at least one channel open) to avoid the pending lsat.token error in the Lightning Terminal.

**Comment by openoms** [comment]
next, torify pool and loop:
```
--loop.server.proxy=                                        The host:port of a SOCKS proxy through which all connections to the loop server will be established over
--pool.proxy=                                               The host:port of a SOCKS proxy through which all connections to the pool server will be established over
```

**Comment by rootzoll** [comment]
OK merged #1994 .. will make now the changes in the SERVICES MENU :)

**Comment by rootzoll** [comment]
In branch `1993-lit` I have added LIT to the menus.

@openoms but to use the cli commands to work there seems to be macaroons missing

```
admin@raspberrypi:~ $ lit-pool
No help topic for '	'
-bash: --macaroonpath=/home/lit/.pool/mainnet/pool.macaroon: No such file or directory
admin@raspberrypi:~ $ lit-loop
No help topic for '	'
-bash: --macaroonpath=/home/lit/.loop/mainnet/loop.macaroon: No such file or directory
```

**Comment by openoms** [comment]
It was meant to be used fromthe `lit` user, bit we could make the aliases work from `admin` as well.

I think switching to `lit` is cleaner and more self-expalanatory. Also consistent with `bos`, `joinmarket` etc

**Comment by rootzoll** [comment]
I also get errors when operating thru lit user: 

```
lit@raspberrypi:~ $ loop terms
[loop] open /home/lit/.loop/mainnet/tls.cert: no such file or directory
```

```
lit@raspberrypi:~ $ pool accounts l
[pool] rpc error: code = Unavailable desc = connection error: desc = "transport: Error while dialing dial tcp [::1]:12010: connect: connection refused"
```

**Comment by openoms** [comment]
hmm, me too. Looking into it now.

Ahh the problem is the line break in the alias, fixing it quick.

**Comment by openoms** [comment]
ok this adds the aliases correctly:
```
  echo "
alias lit-loop=\"loop --rpcserver=localhost:8443 \
  --tlscertpath=/home/lit/.lit/tls.cert \
  --macaroonpath=/home/lit/.loop/${chain}net/loop.macaroon\"
alias lit-pool=\"pool --rpcserver=localhost:8443 \
  --tlscertpath=/home/lit/.lit/tls.cert \
  --macaroonpath=/home/lit/.pool/${chain}net/pool.macaroon\"
alias lit-frcli=\"frcli --rpcserver=localhost:8443 \
  --tlscertpath=/home/lit/.lit/tls.cert \
  --macaroonpath=/home/lit/.faraday/${chain}net/faraday.macaroon\"
" | sudo tee -a /home/lit/.bashrc
```

and they are removed with the `lit` user on uninstall.

**Comment by rootzoll** [comment]
I merged now the `1993-lit` branch into `dev` .. because menu integrations are ok. But I still get those errors: https://github.com/rootzoll/raspiblitz/issues/1993#issuecomment-802009180

**Comment by BtcContributor** [comment]
Hi,
just tried and I cannot reach Lit from Tor onion site but only from local network.
Is this a known issue or is it something related specifically to me?

**Comment by openoms** [comment]
@BtcContributor it works for me. Try restatying Tor:
`sudo systemctl restrt Tor`
and make sure you open the HTTPS:// .onion link.

**Comment by rootzoll** [comment]
Looks good sofar ... Final Test: Upgrade my production node to v1.7RC3 and test LoopIn with LIT.

**Comment by rootzoll** [comment]
LIT seems to run fine on my production node. Closing issue for v1.7

**Comment by maximlomans** [comment]
[frcli] unable to read tls cert (path: /home/lit/.faraday/mainnet/tls.cert): open /home/lit/.faraday/mainnet/tls.cert: no such file or directory 

did a quick copy from lit user 
cp ~/.lit/tls.cert ~/.faraday/mainnet 
but plan to follow openoms echoing advice to admin
